neuraxle/hyperparams/space.py

```diff
@@ -167,13 +157,73 @@ def to_nested_dict_as_ordered_dict_primitive(self) -> OrderedDict:
         return flat_to_nested_dict(self, dict_ctor=OrderedDict)
 
 
-class HyperparameterSpace(HyperparameterSamples):
-    """Wraps an hyperparameter nested dict or flat dict, and offer a few more functions to process
+def nested_dict_to_flat(nested_dict, dict_ctor=OrderedDict):
+    """
+    Convert a nested hyperparameter dictionary to a flat one.
+
+    :param nested_dict: a nested hyperparameter dictionary.
+    :param dict_ctor: ``OrderedDict`` by default. Will use this as a class to create the new returned dict.
+    :return: a flat hyperparameter dictionary.
+    """
+    ret = dict_ctor()
+    for k, v in nested_dict.items():
+        if isinstance(v, dict) or isinstance(v, OrderedDict) or isinstance(v, dict_ctor):
+            _ret = nested_dict_to_flat(v, dict_ctor)
+            for key, val in _ret.items():
+                ret[k + PARAMS_SPLIT_SEQ + key] = val
+        else:
+            ret[k] = v
+    return ret
+
+
+def flat_to_nested_dict(flat_dict, dict_ctor=dict):
+    """
+    Convert a flat hyperparameter dictionary to a nested one.
+
+    :param flat_dict: a flat hyperparameter dictionary.
+    :param dict_ctor: ``OrderedDict`` by default. Will use this as a class to create the new returned dict.
+    :return: a nested hyperparameter dictionary.
+    """
+    pre_ret = dict_ctor()
+    ret = dict_ctor()
+    for k, v in flat_dict.items():
+        k, _, key = k.partition(PARAMS_SPLIT_SEQ)
```

Comment:
Missing argument. Bug: 
```suggestion
        k, _, key = k.partition(separator)
```